packages/workspace/src/core/project-graph/project-graph-models.ts

```diff
@@ -21,16 +21,19 @@ export interface ProjectGraphNode<T extends {} = {}> {
 
 export type ProjectGraphNodeRecords = Record<string, ProjectGraphNode>;
 
-export type AddProjectNode = (node: ProjectGraphNode) => void;
+export type AddProjectNode = (
+  node: ProjectGraphNode,
+  mergeData?: (d: any) => any
+) => void;
 
 export interface ProjectGraphDependency {
-  type: DependencyType;
+  type: DependencyType | string;
```

Comment:
Same as above, why is `string` added here?

Reply:
Like above, the `type` should be `string` since it's open for extension.